### PR TITLE
bugfix/870 - A/C appear to stop taxiing after one was told to change runways

### LINKED ISSUE #870: A/C appear to stop taxiing after one was told to change runways
Browser: Chrome

Airport: KSTL & EIDW tested so far

Approximate play time before issue: 0

#### Issue Description
At St Louis, I told an A/C to `taxi 30R` then changed my mind and said `taxi 30L`. After that, any further A/C told to `taxi 30R` don't appear on the runway. They can be told to `taxi 30L` or any of the other runways, but 30R seems to be blocked after this. There doesn't seem to be any errors in the console related to this.

### PR DESCRIPTION
**Pull request by sam-irl**
Resolves #870.

Fix unusable runway bug after changing one aircraft's departure runway.
Also ensure `taxied` is used in lieu of `taxiied`.

### CODE REVIEW

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 1]
```diff
@@ -645,19 +645,25 @@ export default class AircraftCommander {
         let taxiDestination = data[0];
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftCommander.js]: An you add an empty line above this please?

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 2]
```diff
@@ -645,19 +645,25 @@ export default class AircraftCommander {
...
+        const airport = AirportController.airport_get();
 
         // Set the runway to taxi to
         if (!taxiDestination) {
-            const airport = AirportController.airport_get();
             taxiDestination = airport.departureRunwayModel.name;
         }
 
-        const runway = AirportController.airport_get().getRunway(taxiDestination.toUpperCase());
+        const runway = airport.getRunway(taxiDestination.toUpperCase());
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftCommander.js]: The `.toUpperCase()` should be handled by `.getRunway()`. If it isn’t there already, please add it there.

Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: `.getRunway()` actually gets the runway _model_, not the name, so we don't want to try to do a `.toUpperCase()` on the object. Thus, this should instead be `taxiDestination.name.toUpperCase()`

File: src/assets/scripts/client/aircraft/AircraftCommander.js (hunk 1 above)
Comment (erikquinn) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: done

File: src/assets/scripts/client/aircraft/AircraftCommander.js [hunk 3]
```diff
@@ -556,47 +556,59 @@ export default class AircraftCommander {
...
-        aircraft.fms.setDepartureRunway(runwayModel);
-        runwayModel.addAircraftToQueue(aircraft.id);
-        aircraft.setFlightPhase(FLIGHT_PHASE.TAXI);
+        const readback = aircraftModel.pilot.taxiToRunway(nextRunwayModel, aircraftModel.isDeparture());
+        aircraftModel.taxi_start = taxiStartTime;
 
+        // add aircraft to new runway's queue
+        aircraftModel.fms.departureRunwayModel.addAircraftToQueue(aircraftModel.id);
+
+
```
[the remark this answers is not in the document]
Comment (felixscheffer) [reply src/assets/scripts/client/aircraft/AircraftCommander.js]: done

File: test/aircraft/Pilot/Pilot.spec.js [hunk 4]
```diff
@@ -937,36 +938,17 @@ ava('.sayTargetHeading() returns a message when #headingMode is OFF', (t) => {
...
-});
+ava('.taxiToRunway() returns an error when the aircraft\'s flight phase is not "APRON", "TAXI", or "WAITING"', (t) => {
+    const nonTaxiFlightPhases = _omit(FLIGHT_PHASE, 'APRON', 'TAXI', 'WAITING');
 
-ava('.taxiToRunway() returns an error when flightPhase is not equal to FLIGHT_PHASE.APRON', (t) => {
-    const expectedResult = [false, 'unable to taxi'];
-    const pilot = createPilotFixture();
-    const result = pilot.taxiToRunway(runwayModelMock, true, FLIGHT_PHASE.ARRIVAL);
+    for (const phase in nonTaxiFlightPhases) {
+        console.log(phase);
```
[the remark this answers is not in the document]
Comment (felixscheffer) [reply test/aircraft/Pilot/Pilot.spec.js]: done

**Comment by felixscheffer** [comment]
There is a hidden change in behavior in this PR:
Currently the "taxi" command (without runway) sends the aircraft to the airport's default departure runway.  This PR uses  the departure runway from the FMS instead. 

I dont think we should change the behavior of the "taxi" command with this PR as it's not part of the ticket.

**Comment by felixscheffer** [comment]
I also noticed that we are setting FLIGHT_PHASE.WAITING twice.

1. in `_changeFromTaxiToWaiting()` which is triggered by `GameController.game_timeout` in `taxiToRunway()`
2. in `updateFlightPhase` after `taxi_time` has elapsed

I think we should choose one. Any preferences?

File: src/assets/scripts/client/aircraft/AircraftModel.js [hunk 5]
```diff
@@ -1178,6 +1180,56 @@ export default class AircraftModel {
...
+            return  [false, 'uanble to taxi to runway, we have just landed'];
+        }
+
+        if (!this.fms.isRunwayModelValidForSid(runwayModel)) {
+            this.pilot.cancelDepartureClearance(this);
+        }
+
+        // remove aircraft from previous runway's queue
+        this.fms.departureRunwayModel.removeAircraftFromQueue(this.id);
+
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftModel.js]: Remove empty line

File: src/assets/scripts/client/aircraft/AircraftModel.js [hunk 6]
```diff
@@ -1178,6 +1180,56 @@ export default class AircraftModel {
...
+        this.setFlightPhase(FLIGHT_PHASE.TAXI);
+        this.fms.setDepartureRunway(runwayModel);
+        this.fms.departureRunwayModel.addAircraftToQueue(this.id);
+        this.taxi_start = TimeKeeper.accumulatedDeltaTime;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftModel.js]: Add empty line between method calls and assignments

File: test/aircraft/Pilot/Pilot.spec.js [hunk 7]
```diff
@@ -937,36 +938,17 @@ ava('.sayTargetHeading() returns a message when #headingMode is OFF', (t) => {
...
+        const expectedResult = [false, 'unable to taxi'];
+        const pilot = createPilotFixture();
+        const result = pilot.taxiToRunway(runwayModelMock, false, FLIGHT_PHASE[phase]);
```
[the remark this answers is not in the document]
Comment (erikquinn) [reply test/aircraft/Pilot/Pilot.spec.js]: done

**Pull request merged**